diff --git a/scripts/package-build/linux-kernel/build-linux-firmware.sh b/scripts/package-build/linux-kernel/build-linux-firmware.sh
index 2b1fa7b7..0128d0cd 100755
--- a/scripts/package-build/linux-kernel/build-linux-firmware.sh
+++ b/scripts/package-build/linux-kernel/build-linux-firmware.sh
@@ -48,6 +48,21 @@ mkdir -p "${LINUX_FIRMWARE_BUILD_DIR}"
     ./copy-firmware.sh "${CWD}/${LINUX_FIRMWARE_BUILD_DIR}"
 )
 
+extra_firmwares="mediatek ath11k ath12k"
+
+for fw in $extra_firmwares; do
+    src_dir="${LINUX_FIRMWARE_BUILD_DIR}/${fw}"
+    dest_dir="${VYOS_FIRMWARE_DIR}/lib/firmware"
+
+    if [ -d "$src_dir" ]; then
+        mkdir -p "$dest_dir"
+        cp -a "$src_dir" "$dest_dir/"
+        echo "I: installed entire $fw firmware folder"
+    else
+        echo "W: $fw firmware folder does not exist in $LINUX_FIRMWARE_BUILD_DIR"
+    fi
+done
+
 # Copy firmware file from linux firmware build directory into
 # assembly folder for the vyos-firmware package
 SED_REPLACE="s@${CWD}/${LINUX_FIRMWARE}/@@"
